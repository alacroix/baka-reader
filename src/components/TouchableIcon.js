// @flow

import React from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import appStyle from 'bakareader/src/appStyle';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
});

type PropsType = {
  name: string,
  onPress: () => void,
};

function TouchableIcon(props: PropsType) {
  return (
    <TouchableOpacity activeOpacity={0.5} onPress={props.onPress} style={styles.container}>
      <Icon name={props.name} size={30} color={appStyle.colors.darkGray} />
    </TouchableOpacity>
  );
}

export default TouchableIcon;
