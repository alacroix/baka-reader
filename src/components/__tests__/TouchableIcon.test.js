import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import TouchableIcon from '../TouchableIcon';

describe('TouchableIcon', () => {
  it('should render the touchable icon', () => {
    const tree = renderer.create(
      <TouchableIcon name="fullscreen" onPress={() => {}} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
