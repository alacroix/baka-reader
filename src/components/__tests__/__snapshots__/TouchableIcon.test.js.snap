// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TouchableIcon should render the touchable icon 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#333333",
          "fontSize": 30,
        },
        undefined,
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;
